src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -161,8 +166,69 @@ impl<'value, 'top> BinaryValueWriter_1_1<'value, 'top> {
         Ok(())
     }
 
-    pub fn write_decimal(self, _value: &Decimal) -> IonResult<()> {
-        todo!()
+    pub fn write_decimal(mut self, value: &Decimal) -> IonResult<()> {
+        // Insert a placeholder opcode; we'll overwrite the length nibble with the appropriate value when the encoding
+        // is complete.
+        let opcode_index = self.encoding_buffer.len();
+        self.push_byte(0x60);
```

Comment:
How expensive is it to splice into the array? (Is it common enough for us to even care?) Is there any particular reason we can't calculate the size using `FlexInt::encoded_size_i64` for the exponent and something like `FixedInt::encoded_size_i64` for the coefficient so that we can write the correct length to begin with?

Maybe answering these questions is something that we punt for later if we don't think it's going to be very common.

Reply:
> How expensive is it to splice into the array?

It can (potentially) involve an allocation.

> (Is it common enough for us to even care?)

I don't think I've ever seen a 16+ byte decimal in the wild, only in tests.

>  Is there any particular reason we can't calculate the size using FlexInt::encoded_size_i64 for the exponent and something like FixedInt::encoded_size_i64 for the coefficient so that we can write the correct length to begin with?

We could do this, though we'd need to support writing `FixedInt`s that were `BigInt` under the hood (not just `i64`). Estimating a `BigInt`'s size is weirdly tricky -- going through [the docs](https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html), the only way I could see to do it was to call `to_bytes_le()` (which allocates a new `Vec`) and check its length. This felt similar in cost to just splicing, so I didn't look into it much further.

Since I think this doesn't come up much, I'd like to punt on optimizing it until I find out why I'm wrong. :P